### DISCUSSION TITLE
Can the BigQuery Partition Sensor be updated to sense a list of partition IDs instead of a single partition ID (string)

### DISCUSSION THREAD

**Discussion by noeun-kim** [apache/airflow#36159] [General] [answered]
### Description
I'm looking to work on sensing multiple partition IDs and I'm attempting to modify the relevant class for this purpose.

I believe it's a simple adjustment—changing `partition_id` from a string type to a list type in `BigQueryTablePartitionExistenceSensor`, `BigQueryHook`

But I'm unsure about the modifications needed in `BigQueryTablePartitionExistenceTrigger`
Could you provide some ideas on how to make this adjustment?

### Use case/motivation

I expect to be used as follows
```
from airflow.providers.google.cloud.sensors.bigquery import BigQueryTablePartitionsExistenceSensor

test_sensor = BigQueryTablePartitionsExistenceSensor(
            task_id=f"test_sensor",
            project_id={project_id},
            dataset_id={dataset_id},
            table_id={table_id},
            partition_ids=[{id}, {id}, ...], # new feature
            poke_interval={poke_interval},
            timeout={timeout},
            deferrable={deferrable},
        )
```
And I think the following modifications will be needed for the BigQueryHook class
 ```
class BigQueryHook(GoogleBaseHook, DbApiHook):
    ...
   @GoogleBaseHook.fallback_to_default_project_id
    def table_partitions_exists(
        self, dataset_id: str, table_id: str, partition_ids: list[str], project_id: str
    ) -> bool:
        """Check if date-type partition ids are sequential in GBQ Table
        :param project_id: GCP project id
        :param dataset_id: The name of the dataset.
        :param table_id: The name of the table to check the existence of.
        :param partition_ids: List of partition ids
        """
        table_reference = TableReference(DatasetReference(project_id, dataset_id), table_id)
        try:
            sequential_check = set(partition_ids).issubset(set(
                self.get_client(project_id=project_id).list_partitions(table_reference))
            )
            return sequential_check
        except NotFound:
            return False
```

Here is my idea [commit](https://github.com/noeun-kim/airflow/commit/3c8c1b91cdea55b8dbec5b0266459633c129b77d)

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment, marked as answer]
Just open PR for that and hopefully someone will comment. There is no need to open issue for it (I converted it into a discussion). Generally we do not need to have an issue when you have already something that can start as a draft PR

**Reply by noeun-kim** [reply]
I've created the PR for this issue -> #36201
